src/scenes/Main.ts

```diff
@@ -0,0 +1,384 @@
+                frames: this.anims.generateFrameNumbers(`germ${i}`, { start: 0, end: 1 }),
+                frameRate: 3,
+                repeat: -1,
+            });
+            this.anims.create({
+                key: `germ${i}Splitting`,
+                frames: this.anims.generateFrameNumbers(`germ${i}`, { start: 2, end: 3 }),
+                frameRate: 3,
+                repeat: -1,
+            });
+        });
+
+        countdownText = this.add.text(0, 0, '10.0', { color: '#00ff00' });
+        stageText = this.add.text(0, 0, 'Stage 1', { color: '#00ff00' });
+        stageText.x = constants.width - stageText.width - 20;
+        // dumb hack to make it taller because it doesn't know how to deal with descenders apparently
+        stageText.setFixedSize(stageText.width + 10, stageText.height + 10);
+
+        germsBlue = this.physics.add.group({
+            classType: GermBlue,
+            maxSize: constants.maxGerms,
+            runChildUpdate: true,
+        });
+        germsGreen = this.physics.add.group({
+            classType: GermGreen,
+            maxSize: constants.maxGerms,
+            runChildUpdate: true,
+        });
+        germsOrange = this.physics.add.group({
+            classType: GermOrange,
+            maxSize: constants.maxGerms,
+            runChildUpdate: true,
+        });
+        germsPink = this.physics.add.group({
+            classType: GermPink,
+            maxSize: constants.maxGerms,
+            runChildUpdate: true,
+        });
+
+        germPoolMap.set('GermBlue', germsBlue);
+        germPoolMap.set('GermGreen', germsGreen);
+        germPoolMap.set('GermOrange', germsOrange);
+        germPoolMap.set('GermPink', germsPink);
+
+        allGerms = [germsBlue, germsGreen, germsOrange, germsPink];
+
+        this.addGerms();
+
+        freezeBullets = this.physics.add.group({
+            classType: FreezeBullet,
+            maxSize: 5,
+            runChildUpdate: true,
+        });
+
+        lasers = this.physics.add.group({
+            classType: Laser,
+            maxSize: 2, // In case player fires at the end of a timer and then at the beginning of the next.
+            runChildUpdate: true,
+        });
+
+        //
+        // Player code + controls
+        //
+
+        player = this.physics.add.sprite(constants.width / 2, 50, 'player').setDepth(1).setCircle(16);
+
+        this.input.mouse.disableContextMenu();
+        this.input.on('pointerdown', (pointer: Phaser.Input.Pointer) => {
+            mouseX = pointer.x;
+            mouseY = pointer.y;
+
+            if (pointer.rightButtonDown()) {
+                fireLaser = true;
+            }
+            else {
+                if (clickToRestart) {
+                    this.scene.start('Main');
+                }
+                fireFreeze = true;
+            }
+        });
+
+        //
+        // Collision code
+        //
+
+        for (let i = 0; i < allGerms.length; i++) {
```

Comment:
any reason you're using index iteration instead of for (const germ of allGerms) {...}

Reply:
Because I didn't know the `of` syntax.

Reply:
Actually, it was also to simplify the later second iteration on `allGerms` also, but I've resolved that differently now.